### DISCUSSION TITLE
How to set the Color for a single Char ?

### DISCUSSION THREAD

**Discussion by paule32** [magiblot/tvision#150] [Q&A] [answered]
Hello,
I tried to "writeChar" a character with a TColorAttr Color, but I fail.
```
TColorAttr Color = { 0xffff00, 0x000000 };
writeChar(cursor.x, cursor.y, 'A', Color, 1);
```
The Color will be white on red - and this is not what I expect.
The Baseclass is a TDialog, but with cpBlue.

**Comment by magiblot** [comment, marked as answer]
Hi Jens! Sorry for not replying earlier.

If you look at the `writeChar` function:

https://github.com/magiblot/tvision/blob/0917f04206fcfa1973dc49547af74e7399241899/source/tvision/tvwrite.cpp#L354-L359

You will see that the `color` parameter is a `uchar` (instead of a `TColorAttr`) and that it is passed as an argument to `mapColor`. This means the `color` parameter of `writeChar` is meant to be a color palette index instead of a literal color, which is why you don't get the result you expected.

The "white on red" color you see is the color shown for invalid palette adresses (originating from `TView::errorAttr`).

In this case, in order to get the behaviour you desire, you should either use a `TDrawBuffer`, whose `writeChar` method does take a `TColorAttr` argument, or use a `TScreenCell` and call `writeView`, passing the address of the `TScreenCell` and `count = 1`.
